### ISSUE TITLE
Permissions: Enable non-superusers to query all pg_dist_ tables & sequences

### ISSUE THREAD

**Issue by lfittl** [citusdata/citus#985]
How to reproduce:

```sql
-- Run the following as a non-superuser (e.g. citus user on Citus Cloud)
citus=> SELECT * FROM pg_dist_node;
ERROR:  permission denied for relation pg_dist_node
```

Expected:

```sql
citus=> SELECT * FROM pg_dist_node;
 nodeid | groupid |                     nodename                     | nodeport | noderack | hasmetadata 
--------+---------+--------------------------------------------------+----------+----------+-------------
      1 |       1 | ec2-52-8-98-25.us-west-1.compute.amazonaws.com   |     5432 | default  | f
      2 |       2 | ec2-52-9-166-136.us-west-1.compute.amazonaws.com |     5432 | default  | f
(2 rows)
```

This is a somewhat cosmetic issue, but the real problem happens when you try doing a structural pg_dump, as tools like Rails' migration framework do, which will error out like this:

```
shasta:citus-example-ad-analytics lfittl$ rake db:migrate
pg_dump: [archiver (db)] query failed: ERROR:  permission denied for relation pg_dist_node
pg_dump: [archiver (db)] query was: LOCK TABLE pg_catalog.pg_dist_node IN ACCESS SHARE MODE
rake aborted!
failed to execute:
pg_dump -s -x -O -f /Users/lfittl/Code/citus/citus-example-ad-analytics/db/structure.sql citus

Please check the output above for any errors and make sure that `pg_dump` is installed in your PATH and has proper permissions.


Tasks: TOP => db:structure:dump
(See full trace by running task with --trace)
```

After talking with the team there doesn't seem to be a good reason that we restrict permissions here.

Proposed fix would be the following, which fixes the Rails migration use case:

```sql
GRANT SELECT ON pg_catalog.pg_dist_node TO public;
GRANT SELECT ON pg_catalog.pg_dist_colocation TO public;
GRANT SELECT ON pg_catalog.pg_dist_colocationid_seq TO public;
GRANT SELECT ON pg_catalog.pg_dist_groupid_seq TO public;
GRANT SELECT ON pg_catalog.pg_dist_node_nodeid_seq TO public;
GRANT SELECT ON pg_catalog.pg_dist_shard_placement_placementid_seq TO public;
GRANT SELECT ON pg_catalog.pg_dist_shardid_seq TO public;
```

We'll likely hotfix this in Cloud, so whilst a backport to 6.0 would be helpful, its not necessarily required from that side.

**Comment by anarazel** [comment]
Hi,

On 2016-11-17 16:21:55 -0800, Lukas Fittl wrote:
> Proposed fix would be the following, which fixes the Rails migration use case:
> 
> ```sql
> GRANT SELECT ON pg_catalog.pg_dist_node TO public;
> GRANT SELECT ON pg_catalog.pg_dist_colocation TO public;
> GRANT SELECT ON pg_catalog.pg_dist_colocationid_seq TO public;
> GRANT SELECT ON pg_catalog.pg_dist_groupid_seq TO public;
> GRANT SELECT ON pg_catalog.pg_dist_node_nodeid_seq TO public;
> GRANT SELECT ON pg_catalog.pg_dist_shard_placement_placementid_seq TO public;
> GRANT SELECT ON pg_catalog.pg_dist_shardid_seq TO public;
> ```

I think that makes sense.


> We'll likely hotfix this in Cloud, so whilst a backport to 6.0 would be helpful, its not necessarily required from that side.

I don't think this even remotely warants divergence. That's a bad
pattern we should reduce, not increase.

**Comment by fdr** [comment]
Why do you think Rails migrations not working is not sufficient to warrant such a divergence? @lfittl as we're shipping 6.0 by default, have we lost an important workflow 5.2?

**Comment by anarazel** [comment]
On 2016-11-22 19:00:50 -0800, Daniel Farina wrote:
> Why do you think Rails migrations not working is not sufficient to warrant such a divergence? @lfittl as we're shipping 6.0 by default, have we lost an important workflow 5.2?

This hasn't changed recently, so it's not related to 6.0, and obviously
not urgent.

**Comment by ozgune** [comment]
@onderkalaci / @marcocitus -- I'm marking this as a relatively easy fix and therefore adding the 6.1 milestone to it. If the fix is more involved, could you reply to this issue?

**Comment by Rod-Santos** [comment]
Hey guys. I'm having the behavior in a postgresql version 13 database, in a rockylinux 9. Did these commands solve the problem?
